test/src/main/java/org/apache/accumulo/test/functional/SplitIT.java

```diff
@@ -140,6 +140,8 @@ public void tabletShouldSplit() throws Exception {
       VerifyParams params = new VerifyParams(getClientProps(), table, 100_000);
       TestIngest.ingest(c, params);
       VerifyIngest.verifyIngest(c, params);
+      // ELASTICITY_TODO: Sum of file sizes is zero in
+      // TabletManagementIterator.shouldReturnDueToSplit, so split never happens
```

Comment:
I am looking into this locally.  Added the following to the TabletManagementIterator.

```java
 try {
          computeTabletManagementActions(tm, actions);
        }catch(IllegalStateException ise) {
          LOG.info("decodedRow.size() : {}", decodedRow.size());
          decodedRow.forEach((k2,v2)->LOG.info("{} {}",k2,v2));
          throw ise;
        }
```

When there is a failure I am seeing a tablet that only has a future location. Still looking into the cause of that.

We may want to add some validation of the TabletManagementIterator.seek() to check if all columns or the expected columns are being seeked.  I added some debug to see if something was just seeking the future column and that was not the case.

I also locally merged the latest elasticity changes into this branch.

Reply:
Tracked down two problems, have submitted a fix for one : #3441

I added a lot of debugging and finally figured out what was happening.   I was seeing the tserver return row `row_00000124%ff;` where `%ff;` is binary.  In the manager it was processing `row_00000124%ef;%bf;%bd;` where `%ef;%bf;%bd;` is binary.  This was caused by data going through String which #3441 fixes.  The row the manager was processing did not actually exists.  It would write a future location for this non-existent tablet and that would cause the tablet w/o a prev row.  I am going to submit another little fix where a tablet can not write a future if the tablet does not exists, that should not be able to happen.

Reply:
Opened #3442 with the 2nd fix found while looking into this

Reply:
I merged your changes and this test passes. Thanks for the assist!